File: B-example-solutions/obj-c/01/Assignment-01-03/Assignment-01-03/ViewController.m
//
//  ViewController.m
//  Assignment-01-03
//
//  Created by Philip Dow on 8/27/15.
//  Copyright (c) 2015 Phil Dow. All rights reserved.
//

#import "ViewController.h"

@interface ViewController () <UITextFieldDelegate>

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

// this is a UITextField delegate method
- (BOOL) textFieldShouldReturn:(UITextField *)textField
{    
    // calling resignFirstResponder on the text field
    // and returning YES hides the keyboard
    [textField resignFirstResponder];
    self.label.text = textField.text;
    return YES;
}

@end

File: B-example-solutions/swift/01/Assignment-01-03/Assignment-01-03/ViewController.swift
//
//  ViewController.swift
//  Assignment-01-03
//
//  Created by Philip Dow on 8/27/15.
//  Copyright (c) 2015 Phil Dow. All rights reserved.
//

import UIKit

class ViewController: UIViewController, UITextFieldDelegate {

    @IBOutlet weak var label: UILabel!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

    func textFieldShouldReturn(textField: UITextField) -> Bool {
        textField.resignFirstResponder()
        label.text = textField.text
        return true
    }

}

